ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/ColumnReferenceExtractor.java

```diff
@@ -0,0 +1,61 @@
+package io.confluent.ksql.analyzer;
+
+import io.confluent.ksql.execution.expression.tree.ComparisonExpression;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.LogicalBinaryExpression;
+import io.confluent.ksql.execution.expression.tree.QualifiedColumnReferenceExp;
+import io.confluent.ksql.execution.expression.tree.UnqualifiedColumnReferenceExp;
+import io.confluent.ksql.execution.expression.tree.VisitParentExpressionVisitor;
+import io.confluent.ksql.name.ColumnName;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * The purpose of this class is to extract ColumnNames from an {@link Expression}.
```

Comment:
I want to make sure I'm implementing at least every necessary method to handle pull queries and SPQs, but I'm not very familiar with this part of the codebase and what is/isn't compatible. I'd appreciate if you could take a look at `ExpressionVisitor` and let me know if there are any other methods I'd need to implement , assuming that'd be trivial for you @vcrfxia